Update python_basics_lists_dictionaries.md (#801)
arcus/education_modules: python_basics_lists_dictionaries/python_basics_lists_dictionaries.md

author:   Meredith Lee
email:    leemc@chop.edu

> version: 1.1.1

Reviewer: ```suggestion
version: 1.1.0
```

<lia-keep>
<script type="text/x-sage">

> produce = ['tomato', 'squash', 'apple', 'cucumber', 'peach', 'spinach']

Reviewer 2: ```suggestion
produce = ["tomato", "squash", "apple", "cucumber", "peach", "spinach"]
```

* Lists can have duplicate values. An example might be the list `[3, 1, 4, 1, 5, 9]`, where the number 1 appears twice.

> * List items don't all have to be of the same type (you can have a mix of strings and numbers, for example, or even a list of lists). `[3.14, 'pi', False, ['number', 'string', 'boolean']]` is an example of a list that contains a float, a string, a boolean value, and a list as its elements.

Reviewer 2: ```suggestion
* List items don't all have to be of the same type (you can have a mix of strings and numbers, for example, or even a list of lists). `[3.14, "pi", False, ["number", "string", "boolean"]]` is an example of a list that contains a float, a string, a boolean value, and a list as its elements.
```

     </lia-keep>
     </div>

     The element at index `0` (the first position in the list) has now been removed from `produce`:

     @sage
     <div class="python_link">
     <lia-keep>
     <script type="text/x-sage">

     print(produce)

     </script>
     </lia-keep>
     </div>

   * **`.remove()`** removes a specific element by **value**.

    @sage
    <div class="python_link">
    <lia-keep>
    <script type="text/x-sage">

    produce.remove('apple')
print(produce)

    </script>
    </lia-keep>
    </div>

    Note that the `.remove()` method, unlike `.pop()`, does not itself return any output. If you want to see what happens if you run `print(produce.remove('apple'))`, try changing the code block above (be sure to re-run all of the code blocks above it!).

>   * **`del`** deletes objects, and can be used as another way to remove elements of a list by index. The syntax will be a little different, because it is a [**keyword**](https://realpython.com/python-keywords/), not a list method. It is useful because it allows you to delete list elements in a **slice**. Slicing uses subsetting notation to indicate where the slice begins and ends. The first index listed is **inclusive**, or included in the slice; this is followed by a colon; the second index listed is **exclusive**, or excluded from the slice; finally, these numbers are surrounded by square brackets.

Reviewer: I wonder if `del` should be in a "learn more" box or some other aside because it isn't a method and we really aren't teaching learners how to use it at all (at least not directly in this paragraph), just telling them that there are other ways to do the same thing... maybe even an "other options" box?

Author: I like this idea!

    </lia-keep>
    </div>

  * **`.append()`** adds an element to the end of a list.

    @sage
    <div class="python_link">
    <lia-keep>
    <script type="text/x-sage">

>     produce.append('pear')

Reviewer 2: ```suggestion
    produce.append("pear")
```

print(produce)

    </script>
    </lia-keep>
    </div>

  * **`.insert()`** adds an element to a specific position.

    @sage
    <div class="python_link">
    <lia-keep>
    <script type="text/x-sage">

>     produce.insert(0, 'kale')

Reviewer 2: ```suggestion
    produce.insert(0, "kale")
```

<lia-keep>
<script type="text/x-sage">

> contact_info = {'address' : '123 Puppydog Lane',
>                 'telephone' : '987-654-3210',
>                 'email' : 'myname@email.com'}

Reviewer 2: ```suggestion
contact_info = {"address" : "123 Puppydog Lane",
                "telephone" : "987-654-3210",
                "email" : "myname@email.com"}
```

    <script type="text/x-sage">

    contact_info_copy = contact_info.copy()

> contact_info_copy.pop('telephone')

Reviewer 2: ```suggestion
contact_info_copy.pop("telephone")
```

    <lia-keep>
    <script type="text/x-sage">

>     contact_info['work_email'] = 'myname@company.com'

Reviewer 2: ```suggestion
    contact_info["work_email"] = "myname@company.com"
```

</div>

<div class = "care">
<b style="color: rgb(var(--color-highlight));">A little encouragement...</b><br>

> We've just gone through quite a few useful list and dictionary methods, and you might be concerned about remembering them all! There's no need to memorize them all right now-- it's likely that you'll start to remember them as you use them more, but in the meantime, feel free to refer back to this module whenever you need to. 

Reviewer: ```suggestion
We've just gone through quite a few useful list and dictionary methods, and you might be concerned about remembering them all! There's no need to memorize them all right now-- it's likely that you'll start to remember them as you use them more, but in the meantime, refer back to this module whenever you need to. 
```
This is more about the way I personally feel about the phrase "feel free to" which I use a lot but also sounds like more like permission than an explicit invitation/instruction. Feel free (haha) to accept or ignore this suggestion.

Author: I see what you mean!

2. How would you access all of the countries (the "keys") in the dictionary below? 

     `capital_cities = {'Afghanistan' : 'Kabul', 'Albania' : 'Tirana', 'Algeria' : 'Algiers', 'Andorra' : 'Andorra la Vella'}`

    [[capital_cities.keys()]]
    <script>
  let input = "@'input".trim();
  input == "capital_cities.keys()";
</script>
    ***
    <div class = "answer">

    To return the keys of a dictionary, you use the `.keys()` dictionary method on the `capital_cities` dictionary. So the correct answer is `capital_cities.keys()`.

    </div>
    ***

3. How would you add the the country of Angola and its capital city of Luanda to the dictionary below?

>     `capital_cities = {'Afghanistan' : 'Kabul', 'Albania' : 'Tirana', 'Algeria' : 'Algiers', 'Andorra' : 'Andorra la Vella'}`

Reviewer 2: ```suggestion
    `capital_cities = {"Afghanistan" : "Kabul", "Albania" : "Tirana", "Algeria" : "Algiers", "Andorra" : "Andorra la Vella"}`
```
